src/download-artifact.ts

```diff
@@ -6,10 +6,21 @@ async function run(): Promise<void> {
   try {
     const name = core.getInput(Inputs.Name, {required: false})
     const path = core.getInput(Inputs.Path, {required: false})
+    const artifactFolder = core.getInput(Inputs.ArtifactFolder, {
+      required: false
+    })
+
+    // parse string input to a boolean
+    const isFolderCreated = artifactFolder.toLocaleLowerCase() === 'true'
 
     const artifactClient = artifact.create()
     if (!name) {
-      // download all artifacts
+      /** Download all artifacts at once
+       *
+       * When downloading all artifacts at once, a separate folder gets created for each artifact. There is currently no option to use
```

Comment:
The more I think about the `downloadAllArtifacts` option, the more I don't like it. I think it would have been better to use a single `downloadArtifact` method in addition to a `listArtifacts` method here. Less maintenance in `@actions/artifact` and it would make it easy to add support for things like downloading artifacts via a wildcard pattern or downloading via a list. 

I'll most definitely be revisiting this , but just be aware that we don't have the flexibility to use the new `artifact-folder` option here without any changes on the `@actions/artifact` package side. I've added a note here and in the `README` that when downloading all artifacts, `artifact-folder` is not respected.